apps/dashboard/app/models/script.rb

```diff
@@ -112,4 +123,21 @@ def save
   end
 
   private
+
+  def add_cluster_to_form(form: [], attributes: {}, clusters: [])
+    form.prepend('cluster') unless form.include?('cluster')
+
+    attributes[:cluster] = if clusters.size > 1
+                             {
+                               widget:  'select',
+                               label:   'Cluster',
+                               options: clusters
+                             }
+                           else
+                             {
+                               value: clusters.first.id.to_s,
+                               fixed: true
+                             }
+                           end
+  end
```

Comment:
Similar to above, maybe something like:

```ruby
def add_cluster_to_form(form: [], attributes: {}, clusters: [])
  ensure_cluster_in_form(form)


private

def ensure_cluster_in_form?(form)
  form.prepend('cluster') unless form.include?('cluster')
end

def clusters_gt_one?(clusters)
  clusters.size > 1
end

def cluster_select_attrs(clusters)
    {
    widget: 'select',
    label: 'Cluster',
    options: clusters
  }
end

def fix_cluster_attrs(cluster)
  {
    value: cluster.first.id.to_s,
    fixed: true
  }
end
```
And then use:
```ruby
  attributes[:cluster] = if multiple_clusters?(clusters)
                                           cluster_select_attribute(clusters)
                                        else
                                          fixed_cluster_attribute(clusters.first)
                                        end
```

Reply:
I refactored this a little bit, but I feel like this is a bit overboard. I mean we're allowed to have more than 1 line in a function.

At some point I feel like having to context switch too often is burdensome. Like OK, I'm fine to break some stuff out, but at some point it's cognitively costly to have to look up, for example, what this function does when I feel like using the direct comparison - `clusters.size > 1` is expressive enough.

```
def clusters_gt_one?(clusters)
  clusters.size > 1
end
```

It's just when _everything_ is a one liner, you end up having to look all over the place to understand anything, and you spend a lot of the time looking around and context switching to actually understand what amounts to like ~5 operations that if they were co-located you could have just understood it directly.

Reply:
I get what you mean, however if this practice was followed one would not really need to context switch a great deal because each problem becomes simple. Personally, I like to see a description of what is happening and hide the details. I'm fine with this though.